simple_node.go

```diff
@@ -84,6 +84,24 @@ func (node *SimpleNode) Equals(node2 Node) bool {
 	}
 
 	tag := node2.Tag()
+	//if both Ancestry sources, only check if their _APID is the same
+	if node.Tag().String() == "Source" && tag.String() == "Source" {
```

Comment:
As mentioned in the other PR #325 this needs to be unit tested. Perhaps it's easier for you to pull this out of this diff so you can land the headers fix in isolation?